src/app/shared/components/course-form/course-form.component.ts

```diff
@@ -1,37 +1,63 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
 import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
 import { fas } from '@fortawesome/free-solid-svg-icons';
 import { ButtonText, ButtonType, IconNames } from '@shared/models/button.model';
-import { mockedAuthorsList } from '@shared/mocks/mocks';
 import { Author } from '@shared/models/author.model';
-import { v4 as uuidv4 } from 'uuid';
+import { CoursesStoreService } from '@app/services/courses-store.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { CoursesService } from '@app/services/courses.service';
 
 @Component({
   selector: 'app-course-form',
   templateUrl: './course-form.component.html',
   styleUrls: ['./course-form.component.scss'],
 })
-export class CourseFormComponent {
+export class CourseFormComponent implements OnInit {
   courseForm!: FormGroup;
-  allAuthors: Author[] = [...mockedAuthorsList];
+  isEditMode!: boolean;
+  courseId!: string;
+  allAuthors: Author[] = [];
   courseAuthors: Author[] = [];
   protected readonly ButtonText = ButtonText;
   protected readonly IconNames = IconNames;
+  protected readonly ButtonType = ButtonType;
 
   constructor(
     public fb: FormBuilder,
-    public library: FaIconLibrary
+    public library: FaIconLibrary,
+    private courseStoreService: CoursesStoreService,
+    private courseService: CoursesService,
+    private route: ActivatedRoute,
+    private router: Router
   ) {
     library.addIconPacks(fas);
-    this.buildForm();
+  }
+
+  ngOnInit() {
+    this.route.params.subscribe(params => {
+      if (params['id']) {
+        this.isEditMode = true;
+        this.courseId = params['id'];
+        this.courseStoreService.authors$.subscribe(authors => {
+          this.allAuthors = authors;
+        });
+        this.loadCourseForEdit();
+      } else {
+        this.courseStoreService.getAllAuthors();
```

Comment:
Hmm.. this getAllAuthors method assigns some value to the authors member of the store. On the next line, you subscribe to that. Is this really needed here?